core/src/main/kotlin/org/jetbrains/research/testspark/core/test/TestSuiteParser.kt

```diff
@@ -1,7 +1,14 @@
-package org.jetbrains.research.testspark.core.test.parsers
+package org.jetbrains.research.testspark.core.test
 
+import org.jetbrains.research.testspark.core.test.data.TestCaseGeneratedByLLM
 import org.jetbrains.research.testspark.core.test.data.TestSuiteGeneratedByLLM
 
+data class TestCaseParseResult(
+    val testCase: TestCaseGeneratedByLLM?,
+    val errorMessage: String,
+    val errorOccurred: Boolean,
+)
```

Comment:
_[Just thinking out loud here; maybe it is an overkill, need to discuss]_

We will likely have these "failable results" in other functions/components in the future (even already have them; e.g., in `TestsCompiler`, we have `TestCasesCompilationResult`, which serves a similar purpose). I thought about abstracting it to the well-defined API as follows.

Imagine having the following class that defines a generic result:
```kotlin
data class OperationResult<Payload>(
  val result: Payload?,
  val error: ErrorDetails?,
)

data class ErrorDetails(val message: String)
```
The invariant for the class is: "**either result or error must present, but not both at the same time**".

And then use it here as:
```kotlin
typealias TestCaseParseResult = OperationResult<TestCaseGeneratedByLLM>
// in other places alike
```

@pderakhshanfar, @arksap2002, @Frosendroska What do you think of it?

Reply:
I like your suggestion more, then the current implementation. But it seems, that it takes some time to switch, and anyway it's better to do it in a separate PR

Reply:
I made an [issue](https://github.com/JetBrains-Research/TestSpark/issues/293) about this change